### PR TITLE
Handle Redirect returnUrlQueryString param

### PR DESCRIPTION
**Pull request by jreij**

### CODE REVIEW

File: googlepay/src/main/java/com/adyen/checkout/googlepay/GooglePayComponent.java [hunk 1]
```diff
@@ -87,12 +87,7 @@ private GooglePayParams getGooglePayParams() {
         final Configuration configuration = ((GenericPaymentMethodDelegate) mPaymentMethodDelegate)
                 .getPaymentMethod()
                 .getConfiguration();
-        final String serverGatewayMerchantId;
-        if (configuration == null) {
-            serverGatewayMerchantId = null;
-        } else {
-            serverGatewayMerchantId = configuration.getGatewayMerchantId();
-        }
+        final String serverGatewayMerchantId = (configuration != null) ? configuration.getGatewayMerchantId() : null;
```
Comment (caiofaustino) [inline googlepay/src/main/java/com/adyen/checkout/googlepay/GooglePayComponent.java:90]: HAHAHAHAHAHA, I agree but I think this and the Release notes thing should be it's own PR

File: redirect/build.gradle [hunk 2]
```diff
@@ -49,6 +49,7 @@ dependencies {
     testImplementation "junit:junit:$junit_version"
     androidTestImplementation "androidx.test.ext:junit:$test_ext_version"
     androidTestImplementation "androidx.test.espresso:espresso-core:$espresso_version"
+    testImplementation "org.robolectric:robolectric:$robolectric_version"
```
Comment (caiofaustino) [inline redirect/build.gradle:52]: Is it only me that gets triggered by the order here? 😛

File: redirect/src/test/java/com/adyen/checkout/redirect/RedirectUtilTest.kt [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+import com.google.common.collect.Iterators
+import org.junit.Assert.assertEquals
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.annotation.Config
+
+@RunWith(RobolectricTestRunner::class)
+@Config(manifest = Config.NONE)
+class RedirectUtilTest {
```
Comment (caiofaustino) [inline redirect/src/test/java/com/adyen/checkout/redirect/RedirectUtilTest.kt:22]: 💪

File: redirect/src/main/java/com/adyen/checkout/redirect/RedirectUtil.kt [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+     */
+    @JvmStatic
+    @Throws(CheckoutException::class)
+    fun parseRedirectResult(data: Uri): JSONObject {
+        Logger.d(TAG, "parseRedirectResult - $data")
+
+        val extractedParams = HashMap<String, String>().apply {
+            data.getQueryParameter(PAYLOAD_PARAMETER)?.let { put(PAYLOAD_PARAMETER, it) }
+            data.getQueryParameter(REDIRECT_RESULT_PARAMETER)?.let { put(REDIRECT_RESULT_PARAMETER, it) }
+            data.getQueryParameter(QUERY_STRING_PARAMETER)?.let { put(QUERY_STRING_RESULT, it) }
```
Comment (caiofaustino) [inline redirect/src/main/java/com/adyen/checkout/redirect/RedirectUtil.kt:77]: My understanding of the `returnUrlQueryString` was a bit different. Let me check with the team.

File: redirect/build.gradle (hunk 2 above)
Comment (jreij) [reply redirect/build.gradle:52]: You're right I didn't notice it

File: redirect/src/main/java/com/adyen/checkout/redirect/RedirectUtil.kt [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+        val extractedParams = HashMap<String, String>().apply {
+            data.getQueryParameter(PAYLOAD_PARAMETER)?.let { put(PAYLOAD_PARAMETER, it) }
+            data.getQueryParameter(REDIRECT_RESULT_PARAMETER)?.let { put(REDIRECT_RESULT_PARAMETER, it) }
+            data.getQueryParameter(PAYMENT_RESULT_PARAMETER)?.let { paymentResult ->
+                data.getQueryParameter(MD_PARAMETER)?.let { md ->
+                    put(PAYMENT_RESULT_PARAMETER, paymentResult)
+                    put(MD_PARAMETER, md)
+                }
+            }
+            data.encodedQuery?.let { put(QUERY_STRING_RESULT, it) }
```
Comment (caiofaustino) [inline redirect/src/main/java/com/adyen/checkout/redirect/RedirectUtil.kt:82]: How is this not adding the `QUERY_STRING_RESULT` when one of the other ones is present? From the tests it seems to work but I don't get how.

File: RELEASE_NOTES.md [hunk 6]
```diff
@@ -9,14 +9,4 @@
...
-- 'isValid' now checks both 'isInputValid' and 'isReady' to be true.
-
-## Fixed
-- Handle Intent results if DropInActivity got destroyed.
-- Queue API request if DropInService is not yet bound to DropInActivity
-
-
-
-
+- Return `returnUrlQueryString` from redirect URL.
```
Comment (caiofaustino) [inline RELEASE_NOTES.md:12]: Maybe it's nice to add "for some redirect payment methods like Swish", otherwise this change doesn't mean anything to the merchants as this is mostly an internal change.

File: redirect/src/main/java/com/adyen/checkout/redirect/RedirectUtil.kt (hunk 5 above)
Comment (jreij) [reply redirect/src/main/java/com/adyen/checkout/redirect/RedirectUtil.kt:82]: It does add `QUERY_STRING_RESULT` as long as there are any query params, even if we don't recognize them. However if there are no query params at all, `data.encodedQuery` is null and `QUERY_STRING_RESULT` won't be added.

File: RELEASE_NOTES.md (hunk 6 above)
Comment (jreij) [reply RELEASE_NOTES.md:12]: Or is it better to remove this completely from the release notes?

**Review by caiofaustino — APPROVED** [review]
